2DRove/Assets/Scripts/StateMachines/SpitterFSM/SpitterAggroState.cs

```diff
@@ -5,45 +5,69 @@ public class SpitterAggroState : SpitterBaseState
     private Transform player;
     private Rigidbody2D rb;
     private Animator animator;
-    private bool flipped = false;
 
     public override void EnterState(SpitterStateManager spitter)
     {
         Debug.Log("Spitter Entering Aggo State...");
-    
-        player = GameObject.FindGameObjectWithTag("Player").GetComponent<Transform>(); 
+        // Safe check to ensure player is not null
+        player = GameObject.FindGameObjectWithTag("Player")?.GetComponent<Transform>(); 
+        if (player == null)
+        {
+            // Debug.LogWarning("Player not found!");
+            spitter.SwitchState(spitter.IdleState);
+            return;
+        }
 
         rb = spitter.GetComponent<Rigidbody2D>();
         animator = spitter.GetComponent<Animator>();
     }
 
     public override void UpdateState(SpitterStateManager spitter)
     {
-        Vector2 Direction = (player.position - spitter.transform.position).normalized;
-        rb.AddForce(Direction * 1f);
+        if (player == null)
+        {
+            spitter.SwitchState(spitter.IdleState);
+            return;
+        }
+
+        Vector2 direction = (player.position - spitter.transform.position).normalized;
+        rb.AddForce(direction * 1f);
         animator.SetFloat("velocity", Mathf.Abs(rb.velocity.x));
 
-        // player is moving horizional
-        if (Direction.x != 0)
+        // If not attacking, flip the spitter to face the player
+        if (!animator.GetBool("isAttacking")) 
         {
-            flipped = Direction.x < 0; // player moved left(flipped = true), else player is moving right(flipped = false)
+            // Check the direction to the player and flip if necessary
+            bool shouldFlip = (direction.x > 0 && spitter.transform.localScale.x < 0) || (direction.x < 0 && spitter.transform.localScale.x > 0);
+            if (shouldFlip)
+            {
+                // Flip by scaling in x direction
+                spitter.transform.localScale = new Vector3(-spitter.transform.localScale.x, spitter.transform.localScale.y, spitter.transform.localScale.z);
+            }
         }
-        spitter.transform.rotation = Quaternion.Euler(new Vector3(0f, flipped ? 180 : 0f, 0f));
-
     }
 
     public override void OnCollisionEnter2D(SpitterStateManager spitter, Collision2D other)
     {
-
+        
     }
-    public override void OnTriggerStay2D(SpitterStateManager spitter, Collider2D other)
+    
+    public override void OnTriggerStay2D(SpitterStateManager spitter, Collider2D other) 
     {
-        // if (other.tag("Player")) // Used CompareTag for optimized performance
-
-        if (other.CompareTag("Player"))
+        // Check for null reference before attempting to switch state
+        if (player != null && other.gameObject == player.gameObject)
         {
             spitter.SwitchState(spitter.AttackState);
         }
     }
 
+    public override void EventTrigger(SpitterStateManager spitter)
```

Comment:
The original design of the EventTrigger is to be called in animation events, during the first frame of the hit animation, it should call EventTrigger, which EventTrigger will swap state to HitState. However, I believe you used TakeDamage which is completely fine